### PR TITLE
Add integration test for Retroactive default StorageClass assignement

### PR DESCRIPTION
**Pull request by RomanBednar**
Add integration test for v1.25 feature - Retroactive default StorageClass assignement (alpha)

KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/3333-reconcile-default-storage-class
Enhancement issue: https://github.com/kubernetes/enhancements/issues/3333
The main feature PR: https://github.com/kubernetes/kubernetes/pull/111467

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Required for alpha.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/volume/persistent_volumes_test.go [hunk 1]
```diff
@@ -1036,6 +1040,196 @@ func TestPersistentVolumeMultiPVsDiffAccessModes(t *testing.T) {
...
+	_, err = testClient.CoreV1().PersistentVolumes().Create(context.TODO(), pv4, metav1.CreateOptions{})
+	if err != nil {
+		t.Errorf("Failed to create PersistentVolume: %v", err)
+	}
+
+	pvc4 := createPVC("pvc-4", ns.Name, "5G", []v1.PersistentVolumeAccessMode{v1.ReadWriteMany}, "")
+	_, err = testClient.CoreV1().PersistentVolumeClaims(ns.Name).Create(context.TODO(), pvc4, metav1.CreateOptions{})
+	if err != nil {
+		t.Errorf("Failed to create PersistentVolumeClaim: %v", err)
+	}
```
Comment (jsafrane) [inline test/integration/volume/persistent_volumes_test.go:1213]: The test should create PVC first and PV after that, to check that if PV does not exist the PVC still does not get the default SC.

**Comment by jsafrane** [comment]
/retest
"Pod got deleted unexpectedly"

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

**Comment by jsafrane** [comment]
/retest
pull-kubernetes-verify complains about generated openapi spec changes, but this PR does not touch it.

**Comment by dims** [comment]
The verify job will only pass after https://github.com/kubernetes/kubernetes/pull/111784 gets merged

**Comment by dims** [comment]
https://github.com/kubernetes/kubernetes/pull/111784 has merged

/retest

**Pull request merged**